tests/tests/device.rs

```diff
@@ -776,3 +776,24 @@ fn vs_main() -> @builtin(position) vec4<f32> {
             ],
         });
     });
+
+#[gpu_test]
+static POLL_DESTROYED_DEVICE: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(TestParameters::default())
+    .run_sync(|ctx| {
+        // Create a resource that needs to be cleaned up…
+        let _buffer = ctx.device.create_buffer(&wgt::BufferDescriptor {
+            label: None,
+            size: 32,
+            usage: wgt::BufferUsages::MAP_READ,
+            mapped_at_creation: false,
+        });
+        // …but don't do anything with it.
+
+        ctx.device.destroy();
+
+        // We now have an empty submission queue, and the device will clean up `_buffer` on the
+        // next `poll` operation.
+
+        ctx.instance.poll_all(true);
```

Comment:
issue: This doesn't exercise textures, as [@jimblandy's](https://github.com/gfx-rs/wgpu/issues/5471#issuecomment-2030503115) did. We should just use his instead.